pkg/options/network/flag.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file contains functions used to implement the '--template-dir' command line option.
+
+package network
+
+import (
+	"os"
+
+	"github.com/spf13/pflag"
+
+	"github.com/openshift/rosa/pkg/helper"
+)
+
+// AddFlag adds the template-dir flag to the given set of command line flags.
+func AddTemplateFlag(flags *pflag.FlagSet) {
+	flags.StringVar(
+		&templateDir,
+		"template-dir",
+		"",
+		"Use a specific template directory, overriding the TEMPLATE_DIR environment variable.",
+	)
+}
+
+// TemplateDir returns a string with the name of the TEMPLATE_DIR being used.
+func TemplateDir() string {
```

Comment:
@den-rgb This logic should be inside your `NewNetworkUserOptions` function with the template directory as a field on the `NetworkUserOptions` struct

So in the `NewNetworkUserOptions` function:

1. Initialise the options struct
2. Check to see if the TEMPLATE_DIR environment variable is set, if yes, use this for the value of the options.templateDir field, otherwise use the defaut
3. Bind a flag to the field allowing the user to override it via the `--template-dir` CLI parameter

That will avoid the introduction of the `templateDir` global variable here and keep things consistent. You can then update your command to read from `opts.templateDir` when trying to load the templates.